integration-tests/src/test/java/com/redhat/parodos/flows/SimpleRestartWorkFlowTest.java

```diff
@@ -0,0 +1,507 @@
+				namespaceWorkFlowTask, sslCertificationWorkFlowTask));
+		workFlowResponseDTO = workflowApi.execute(workFlowRequestDTO);
+
+		assertNotNull("There is no valid WorkFlowExecutionId", workFlowResponseDTO.getWorkFlowExecutionId());
+		assertEquals(WorkStatusEnum.IN_PROGRESS, workFlowResponseDTO.getWorkStatus());
+		log.info("Onboarding workflow execution id: {}", workFlowResponseDTO.getWorkFlowExecutionId());
+
+		workFlowStatusResponseDTO = WorkFlowServiceUtils.waitWorkflowStatusAsync(workflowApi,
+				workFlowResponseDTO.getWorkFlowExecutionId(), WorkFlowStatusResponseDTO.StatusEnum.FAILED);
+
+		assertNotNull(workFlowStatusResponseDTO);
+		assertNotNull(workFlowStatusResponseDTO.getWorkFlowExecutionId());
+		assertEquals(WorkFlowStatusResponseDTO.StatusEnum.FAILED, workFlowStatusResponseDTO.getStatus());
+		log.info("Onboarding workflow execution completed with status {}", workFlowStatusResponseDTO.getStatus());
+
+		log.info("Restarting the onboarding WorkFlow");
+		WorkFlowExecutionResponseDTO restartedWorkFlowResponseDTO = workflowApi
+				.restartWorkFlow(workFlowResponseDTO.getWorkFlowExecutionId());
+		assertEquals(WorkStatusEnum.IN_PROGRESS, restartedWorkFlowResponseDTO.getWorkStatus());
+		assertNotNull(restartedWorkFlowResponseDTO.getWorkFlowExecutionId());
+		log.info("restarted workflow submitted successfully with response: {}", restartedWorkFlowResponseDTO);
+		assertNotEquals(workFlowResponseDTO.getWorkFlowExecutionId(),
+				restartedWorkFlowResponseDTO.getWorkFlowExecutionId());
+		WorkFlowStatusResponseDTO restartedWorkFlowStatusResponseDTO = WorkFlowServiceUtils.waitWorkflowStatusAsync(
+				workflowApi, restartedWorkFlowResponseDTO.getWorkFlowExecutionId(),
+				WorkFlowStatusResponseDTO.StatusEnum.FAILED);
+
+		assertNotNull(restartedWorkFlowStatusResponseDTO);
+		List<String> works = workFlowStatusResponseDTO.getWorks().stream().map(WorkStatusResponseDTO::getName)
+				.collect(Collectors.toList());
+		List<String> restartedWorks = restartedWorkFlowStatusResponseDTO.getWorks().stream()
+				.map(WorkStatusResponseDTO::getName).collect(Collectors.toList());
+		assertEquals(works, restartedWorks);
+		assertNotNull(restartedWorkFlowStatusResponseDTO.getWorkFlowExecutionId());
+		assertEquals(WorkFlowStatusResponseDTO.StatusEnum.FAILED, restartedWorkFlowStatusResponseDTO.getStatus());
+		log.info("Restarted onboarding workflow execution completed with status {}",
+				restartedWorkFlowStatusResponseDTO.getStatus());
+
+		WorkFlowStatusResponseDTO originalWorkflowStatus = workflowApi
+				.getStatus(workFlowResponseDTO.getWorkFlowExecutionId());
+		assertEquals(1, originalWorkflowStatus.getRestartedCount().intValue());
+		WorkFlowStatusResponseDTO restartedWorkflowStatus = workflowApi
+				.getStatus(restartedWorkFlowStatusResponseDTO.getWorkFlowExecutionId());
+		assertEquals(workFlowResponseDTO.getWorkFlowExecutionId(), restartedWorkflowStatus.getOriginalExecutionId());
+		Thread.sleep(2000);
```

Comment:
I put that in order for the next test `runRestartWorkFlowComplexMultipleTimes` to be successful, it seems that, at least locally, parodos does not handle well several executions in a small window.

Maybe because the rollback/fallback of the failed workflow is blocking something?

I have that in the test logs:
```
[main] INFO com.redhat.parodos.flows.SimpleRestartWorkFlowTest - ******** Running The Complex workFlow multiple times ********
[main] INFO com.redhat.parodos.sdkutils.WorkFlowServiceUtils - serverIp is: localhost, serverPort is 8080. Set BasePath to http://localhost:8080
[main] INFO com.redhat.parodos.sdkutils.WorkFlowServiceUtils - Get all available projects
[main] INFO com.redhat.parodos.flows.SimpleRestartWorkFlowTest - Running the Assessment to see what WorkFlows are eligible for this situation:
[main] INFO com.redhat.parodos.flows.SimpleRestartWorkFlowTest - workflow submitted successfully with response: class WorkFlowExecutionResponseDTO {
    workFlowExecutionId: 728dce20-fe54-43d8-8b43-a639aa243391
    workStatus: IN_PROGRESS
}
[main] INFO com.redhat.parodos.flows.SimpleRestartWorkFlowTest - The Following Option Is Available: complexWorkFlow
[main] INFO com.redhat.parodos.flows.SimpleRestartWorkFlowTest - Running the onboarding WorkFlow
[main] INFO com.redhat.parodos.flows.SimpleRestartWorkFlowTest - executes 3 tasks in Parallel with a WorkFlowChecker
[main] INFO com.redhat.parodos.flows.SimpleRestartWorkFlowTest - Onboarding workflow id 0836783b-601e-4541-9e52-5936de2355b9
[main] INFO com.redhat.parodos.flows.SimpleRestartWorkFlowTest - Onboarding workflow name complexWorkFlow
[main] INFO com.redhat.parodos.flows.SimpleRestartWorkFlowTest - Onboarding workflow execution id: 4b358d0d-4253-424a-a0d8-5bd0c20b977b
```
I have that in the workflow-service log:
```
023-06-27 11:32:09.993  INFO 78685 --- [        task-20] c.r.p.w.e.service.WorkFlowExecutorImpl   : Execute workflow complexWorkFlow (ID: 4b358d0d-4253-424a-a0d8-5bd0c20b977b)
2023-06-27 11:32:09.993  INFO 78685 --- [        task-20] c.r.p.w.engine.WorkFlowEngineImpl        : Running workflow 'complexWorkFlow'
2023-06-27 11:32:09.993  INFO 78685 --- [        task-20] c.r.p.w.e.a.WorkFlowExecutionAspect      : Before invoking execute() on workflow: complexWorkFlow with workContext: context={WORKFLOW_EXECUTION_SUBWORKFLOWTHREE_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_ID=4b358d0d-4253-424a-a0d8-5bd0c20b977b, WORKFLOW_EXECUTION_SUBWORKFLOWONE_PARENT_WORKFLOW=subWorkFlowTwo, USER_ID=b0f58378-a192-43f9-945d-00eb132f11c3, WORKFLOW_EXECUTION_SUBWORKFLOWTWO_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_EXECUTION_SUBWORKFLOWFOUR_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_ARGUMENTS={domainName=api.com, ipAddress=127.0.0.1}, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_ARGUMENTS={adGroups=adGroupsVALUE, userId=test-user-id}, WORKFLOW_EXECUTION_ADGROUPSWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_EXECUTION_NAMESPACEWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowTwo, WORKFLOW_EXECUTION_SINGLESIGNONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, WORKFLOW_DEFINITION_NAME=complexWorkFlow, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_ARGUMENTS={projectId=e068417b-37e5-404b-9009-2a2899917a18}, WORKFLOW_EXECUTION_LOADBALANCERWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, PROJECT_ID=3c6cf116-0592-42e4-96e3-15a8a1f7b6ea}}
2023-06-27 11:32:09.995  INFO 78685 --- [        task-20] c.r.p.w.e.a.WorkFlowExecutionAspect      : Before invoking execute() on workflow: subWorkFlowThree with workContext: context={WORKFLOW_EXECUTION_SUBWORKFLOWTHREE_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_ID=4b358d0d-4253-424a-a0d8-5bd0c20b977b, WORKFLOW_EXECUTION_SUBWORKFLOWONE_PARENT_WORKFLOW=subWorkFlowTwo, USER_ID=b0f58378-a192-43f9-945d-00eb132f11c3, WORKFLOW_EXECUTION_SUBWORKFLOWTWO_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_EXECUTION_SUBWORKFLOWFOUR_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_ARGUMENTS={domainName=api.com, ipAddress=127.0.0.1}, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_ARGUMENTS={adGroups=adGroupsVALUE, userId=test-user-id}, WORKFLOW_EXECUTION_ADGROUPSWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_EXECUTION_NAMESPACEWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowTwo, WORKFLOW_EXECUTION_SINGLESIGNONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, WORKFLOW_DEFINITION_NAME=complexWorkFlow, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_ARGUMENTS={projectId=e068417b-37e5-404b-9009-2a2899917a18}, WORKFLOW_EXECUTION_LOADBALANCERWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, PROJECT_ID=3c6cf116-0592-42e4-96e3-15a8a1f7b6ea}}
2023-06-27 11:32:09.998  INFO 78685 --- [pool-2-thread-2] c.r.p.w.e.a.WorkFlowTaskExecutionAspect  : Before invoking execute() on workflow task name: sslCertificationWorkFlowTask
2023-06-27 11:32:09.998  INFO 78685 --- [pool-2-thread-1] c.r.p.w.e.a.WorkFlowExecutionAspect      : Before invoking execute() on workflow: subWorkFlowTwo with workContext: context={WORKFLOW_EXECUTION_SUBWORKFLOWTHREE_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_ID=4b358d0d-4253-424a-a0d8-5bd0c20b977b, WORKFLOW_EXECUTION_SUBWORKFLOWONE_PARENT_WORKFLOW=subWorkFlowTwo, USER_ID=b0f58378-a192-43f9-945d-00eb132f11c3, WORKFLOW_EXECUTION_SUBWORKFLOWTWO_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_EXECUTION_SUBWORKFLOWFOUR_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_ARGUMENTS={domainName=api.com, ipAddress=127.0.0.1}, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_ARGUMENTS={adGroups=adGroupsVALUE, userId=test-user-id}, WORKFLOW_EXECUTION_ADGROUPSWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_EXECUTION_NAMESPACEWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowTwo, WORKFLOW_EXECUTION_SINGLESIGNONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, WORKFLOW_DEFINITION_NAME=complexWorkFlow, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_ARGUMENTS={projectId=e068417b-37e5-404b-9009-2a2899917a18}, WORKFLOW_EXECUTION_LOADBALANCERWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, PROJECT_ID=3c6cf116-0592-42e4-96e3-15a8a1f7b6ea}}
2023-06-27 11:32:10.000  INFO 78685 --- [   scheduling-1] c.r.p.w.e.a.WorkFlowExecutionAspect      : Before invoking execute() on workflow: sslCertificationApprovalWorkFlowChecker with workContext: context={WORKFLOW_EXECUTION_SUBWORKFLOWTHREE_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_ID=3783a426-caef-4ccf-bcc8-583ab02202c8, WORKFLOW_EXECUTION_SUBWORKFLOWONE_PARENT_WORKFLOW=subWorkFlowTwo, USER_ID=b0f58378-a192-43f9-945d-00eb132f11c3, WORKFLOW_EXECUTION_SUBWORKFLOWTWO_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_EXECUTION_SUBWORKFLOWFOUR_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_ARGUMENTS={domainName=api.com, ipAddress=127.0.0.1}, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_STATUS=FAILED, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_ARGUMENTS={adGroups=adGroupsVALUE}, WORKFLOW_EXECUTION_ADGROUPSWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_TASK_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_NAMESPACEWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowTwo, WORKFLOW_EXECUTION_SINGLESIGNONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, WORKFLOW_DEFINITION_NAME=complexWorkFlow, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_ARGUMENTS={projectId=48def7d9-20aa-4e73-a393-70d350843317}, WORKFLOW_EXECUTION_LOADBALANCERWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, PROJECT_ID=3c6cf116-0592-42e4-96e3-15a8a1f7b6ea}}
2023-06-27 11:32:10.000  INFO 78685 --- [pool-2-thread-1] c.r.p.w.e.a.WorkFlowExecutionAspect      : Before invoking execute() on workflow: subWorkFlowOne with workContext: context={WORKFLOW_EXECUTION_SUBWORKFLOWTHREE_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_ID=4b358d0d-4253-424a-a0d8-5bd0c20b977b, WORKFLOW_EXECUTION_SUBWORKFLOWONE_PARENT_WORKFLOW=subWorkFlowTwo, USER_ID=b0f58378-a192-43f9-945d-00eb132f11c3, WORKFLOW_EXECUTION_SUBWORKFLOWTWO_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_EXECUTION_SUBWORKFLOWFOUR_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_ARGUMENTS={domainName=api.com, ipAddress=127.0.0.1}, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_ARGUMENTS={adGroups=adGroupsVALUE, userId=test-user-id}, WORKFLOW_EXECUTION_ADGROUPSWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_EXECUTION_NAMESPACEWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowTwo, WORKFLOW_EXECUTION_SINGLESIGNONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, WORKFLOW_DEFINITION_NAME=complexWorkFlow, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_ARGUMENTS={projectId=e068417b-37e5-404b-9009-2a2899917a18}, WORKFLOW_EXECUTION_LOADBALANCERWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, PROJECT_ID=3c6cf116-0592-42e4-96e3-15a8a1f7b6ea}}
2023-06-27 11:32:10.001  INFO 78685 --- [pool-2-thread-2] c.r.p.e.c.t.SslCertificationWorkFlowTask : SslCertificationWorkFlowTask
2023-06-27 11:32:10.002  INFO 78685 --- [pool-1-thread-1] c.r.p.w.e.a.WorkFlowTaskExecutionAspect  : Before invoking execute() on workflow task name: adGroupsWorkFlowTask
2023-06-27 11:32:10.002  INFO 78685 --- [pool-1-thread-2] c.r.p.w.e.a.WorkFlowTaskExecutionAspect  : Before invoking execute() on workflow task name: splunkMonitoringWorkFlowTask
2023-06-27 11:32:10.002  INFO 78685 --- [   scheduling-1] c.r.p.w.e.a.WorkFlowTaskExecutionAspect  : Before invoking execute() on workflow task name: sslCertificationApprovalWorkFlowCheckerTask
2023-06-27 11:32:10.002  INFO 78685 --- [pool-2-thread-2] c.r.p.w.e.a.WorkFlowTaskExecutionAspect  : Schedule workflow checker: sslCertificationApprovalWorkFlowChecker to run per cron expression: */5 * * * * ?
2023-06-27 11:32:10.002  INFO 78685 --- [pool-2-thread-2] c.r.p.w.e.s.WorkFlowSchedulerServiceImpl : Workflow: sslCertificationApprovalWorkFlowChecker is already scheduled for project: 3c6cf116-0592-42e4-96e3-15a8a1f7b6ea and user: b0f58378-a192-43f9-945d-00eb132f11c3!
2023-06-27 11:32:10.004  INFO 78685 --- [pool-1-thread-1] c.r.p.e.c.task.AdGroupsWorkFlowTask      : AdGroupsWorkFlowTask
2023-06-27 11:32:10.004  INFO 78685 --- [pool-1-thread-2] c.r.p.e.c.t.SplunkMonitoringWorkFlowTask : SplunkMonitoringWorkFlowTask
2023-06-27 11:32:10.005  INFO 78685 --- [   scheduling-1] CertificationApprovalWorkFlowCheckerTask : SslCertificationApprovalWorkFlowCheckerTask
2023-06-27 11:32:10.005  INFO 78685 --- [pool-2-thread-1] c.r.p.w.e.a.WorkFlowExecutionAspect      : Workflow subWorkFlowOne is COMPLETED!
2023-06-27 11:32:10.005  INFO 78685 --- [   scheduling-1] c.r.p.w.e.a.WorkFlowExecutionAspect      : Workflow sslCertificationApprovalWorkFlowChecker is COMPLETED!
2023-06-27 11:32:10.006  INFO 78685 --- [pool-2-thread-1] c.r.p.w.e.a.WorkFlowTaskExecutionAspect  : Before invoking execute() on workflow task name: namespaceWorkFlowTask
2023-06-27 11:32:10.006  INFO 78685 --- [   scheduling-1] r.p.w.e.a.CheckerWorkFlowPostInterceptor : Stop workflow checker: sslCertificationApprovalWorkFlowChecker schedule
2023-06-27 11:32:10.006  INFO 78685 --- [   scheduling-1] c.r.p.w.e.s.WorkFlowSchedulerServiceImpl : Stopping workflow: sslCertificationApprovalWorkFlowChecker for project: 3c6cf116-0592-42e4-96e3-15a8a1f7b6ea and user: b0f58378-a192-43f9-945d-00eb132f11c3
2023-06-27 11:32:10.008  INFO 78685 --- [        task-21] c.r.p.w.e.service.WorkFlowExecutorImpl   : Execute workflow complexWorkFlow (ID: 3783a426-caef-4ccf-bcc8-583ab02202c8)
2023-06-27 11:32:10.008  INFO 78685 --- [        task-21] c.r.p.w.engine.WorkFlowEngineImpl        : Running workflow 'complexWorkFlow'
2023-06-27 11:32:10.008  INFO 78685 --- [        task-21] c.r.p.w.e.a.WorkFlowExecutionAspect      : Before invoking execute() on workflow: complexWorkFlow with workContext: context={WORKFLOW_EXECUTION_SUBWORKFLOWTHREE_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_ID=3783a426-caef-4ccf-bcc8-583ab02202c8, WORKFLOW_EXECUTION_SUBWORKFLOWONE_PARENT_WORKFLOW=subWorkFlowTwo, USER_ID=b0f58378-a192-43f9-945d-00eb132f11c3, WORKFLOW_EXECUTION_SUBWORKFLOWTWO_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONAPPROVALWORKFLOWCHECKERTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_SUBWORKFLOWFOUR_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_ARGUMENTS={domainName=api.com, ipAddress=127.0.0.1}, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_STATUS=FAILED, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_ARGUMENTS={adGroups=adGroupsVALUE}, WORKFLOW_EXECUTION_ADGROUPSWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_TASK_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_NAMESPACEWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowTwo, WORKFLOW_EXECUTION_SINGLESIGNONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, WORKFLOW_DEFINITION_NAME=complexWorkFlow, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_ARGUMENTS={projectId=48def7d9-20aa-4e73-a393-70d350843317}, WORKFLOW_EXECUTION_LOADBALANCERWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, PROJECT_ID=3c6cf116-0592-42e4-96e3-15a8a1f7b6ea}}
2023-06-27 11:32:10.008  INFO 78685 --- [pool-2-thread-1] c.r.p.e.c.task.NamespaceWorkFlowTask     : NamespaceWorkFlowTask
2023-06-27 11:32:10.009  INFO 78685 --- [        task-21] c.r.p.w.e.a.WorkFlowExecutionAspect      : Before invoking execute() on workflow: subWorkFlowThree with workContext: context={WORKFLOW_EXECUTION_SUBWORKFLOWTHREE_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_ID=3783a426-caef-4ccf-bcc8-583ab02202c8, WORKFLOW_EXECUTION_SUBWORKFLOWONE_PARENT_WORKFLOW=subWorkFlowTwo, USER_ID=b0f58378-a192-43f9-945d-00eb132f11c3, WORKFLOW_EXECUTION_SUBWORKFLOWTWO_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONAPPROVALWORKFLOWCHECKERTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_SUBWORKFLOWFOUR_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_ARGUMENTS={domainName=api.com, ipAddress=127.0.0.1}, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_STATUS=FAILED, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_ARGUMENTS={adGroups=adGroupsVALUE}, WORKFLOW_EXECUTION_ADGROUPSWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_TASK_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_NAMESPACEWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowTwo, WORKFLOW_EXECUTION_SINGLESIGNONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, WORKFLOW_DEFINITION_NAME=complexWorkFlow, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_ARGUMENTS={projectId=48def7d9-20aa-4e73-a393-70d350843317}, WORKFLOW_EXECUTION_LOADBALANCERWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, PROJECT_ID=3c6cf116-0592-42e4-96e3-15a8a1f7b6ea}}
2023-06-27 11:32:10.009  INFO 78685 --- [pool-2-thread-1] c.r.p.w.e.a.WorkFlowTaskExecutionAspect  : Schedule workflow checker: namespaceApprovalWorkFlowChecker to run per cron expression: */5 * * * * ?
2023-06-27 11:32:10.009  INFO 78685 --- [pool-2-thread-1] c.r.p.w.e.s.WorkFlowSchedulerServiceImpl : Scheduling workflow: namespaceApprovalWorkFlowChecker for project: 3c6cf116-0592-42e4-96e3-15a8a1f7b6ea and user: b0f58378-a192-43f9-945d-00eb132f11c3 to be executed following cron expression: */5 * * * * ?
2023-06-27 11:32:10.009  INFO 78685 --- [pool-2-thread-1] c.r.p.workflows.workflow.SequentialFlow  : Work unit "namespaceWorkFlowTask" returned with WorkStatus = "PENDING", skipping subsequent work units
2023-06-27 11:32:10.009  INFO 78685 --- [pool-2-thread-1] c.r.p.w.e.a.WorkFlowExecutionAspect      : Workflow subWorkFlowTwo is PENDING!
2023-06-27 11:32:10.010  INFO 78685 --- [pool-2-thread-1] entInfrastructureWorkFlowPostInterceptor : workflow: subWorkFlowTwo has a pending/running checker: checker is pending in execution: 4b358d0d-4253-424a-a0d8-5bd0c20b977b
2023-06-27 11:32:10.010  INFO 78685 --- [        task-20] c.r.p.w.e.a.WorkFlowExecutionAspect      : Workflow subWorkFlowThree is PENDING!
2023-06-27 11:32:10.011  INFO 78685 --- [pool-2-thread-2] c.r.p.w.e.a.WorkFlowTaskExecutionAspect  : Before invoking execute() on workflow task name: sslCertificationWorkFlowTask
2023-06-27 11:32:10.011  INFO 78685 --- [        task-20] entInfrastructureWorkFlowPostInterceptor : workflow: subWorkFlowThree has a pending/running checker: checker is pending in execution: 4b358d0d-4253-424a-a0d8-5bd0c20b977b
2023-06-27 11:32:10.011  INFO 78685 --- [pool-2-thread-1] c.r.p.w.e.a.WorkFlowExecutionAspect      : Before invoking execute() on workflow: subWorkFlowTwo with workContext: context={WORKFLOW_EXECUTION_SUBWORKFLOWTHREE_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_ID=3783a426-caef-4ccf-bcc8-583ab02202c8, WORKFLOW_EXECUTION_SUBWORKFLOWONE_PARENT_WORKFLOW=subWorkFlowTwo, USER_ID=b0f58378-a192-43f9-945d-00eb132f11c3, WORKFLOW_EXECUTION_SUBWORKFLOWTWO_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONAPPROVALWORKFLOWCHECKERTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_SUBWORKFLOWFOUR_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_ARGUMENTS={domainName=api.com, ipAddress=127.0.0.1}, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_STATUS=FAILED, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_ARGUMENTS={adGroups=adGroupsVALUE}, WORKFLOW_EXECUTION_ADGROUPSWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_TASK_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_NAMESPACEWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowTwo, WORKFLOW_EXECUTION_SINGLESIGNONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, WORKFLOW_DEFINITION_NAME=complexWorkFlow, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_ARGUMENTS={projectId=48def7d9-20aa-4e73-a393-70d350843317}, WORKFLOW_EXECUTION_LOADBALANCERWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, PROJECT_ID=3c6cf116-0592-42e4-96e3-15a8a1f7b6ea}}
2023-06-27 11:32:10.011  INFO 78685 --- [        task-20] c.r.p.workflows.workflow.SequentialFlow  : Work unit "subWorkFlowThree" returned with WorkStatus = "IN_PROGRESS", skipping subsequent work units
2023-06-27 11:32:10.011  INFO 78685 --- [        task-20] c.r.p.w.e.a.WorkFlowExecutionAspect      : Workflow complexWorkFlow is IN_PROGRESS!
2023-06-27 11:32:10.012  INFO 78685 --- [pool-2-thread-1] c.r.p.w.e.a.WorkFlowExecutionAspect      : Before invoking execute() on workflow: subWorkFlowOne with workContext: context={WORKFLOW_EXECUTION_SUBWORKFLOWTHREE_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_ID=3783a426-caef-4ccf-bcc8-583ab02202c8, WORKFLOW_EXECUTION_SUBWORKFLOWONE_PARENT_WORKFLOW=subWorkFlowTwo, USER_ID=b0f58378-a192-43f9-945d-00eb132f11c3, WORKFLOW_EXECUTION_SUBWORKFLOWTWO_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONAPPROVALWORKFLOWCHECKERTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_SUBWORKFLOWFOUR_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_ARGUMENTS={domainName=api.com, ipAddress=127.0.0.1}, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_STATUS=FAILED, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_ARGUMENTS={adGroups=adGroupsVALUE}, WORKFLOW_EXECUTION_ADGROUPSWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_TASK_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_NAMESPACEWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowTwo, WORKFLOW_EXECUTION_SINGLESIGNONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, WORKFLOW_DEFINITION_NAME=complexWorkFlow, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_ARGUMENTS={projectId=48def7d9-20aa-4e73-a393-70d350843317}, WORKFLOW_EXECUTION_LOADBALANCERWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, PROJECT_ID=3c6cf116-0592-42e4-96e3-15a8a1f7b6ea}}
2023-06-27 11:32:10.013  INFO 78685 --- [pool-2-thread-2] c.r.p.w.e.a.WorkFlowTaskExecutionAspect  : skipping task: sslCertificationWorkFlowTask with status COMPLETED
2023-06-27 11:32:10.014  INFO 78685 --- [pool-1-thread-2] c.r.p.w.e.a.WorkFlowTaskExecutionAspect  : Before invoking execute() on workflow task name: adGroupsWorkFlowTask
2023-06-27 11:32:10.014  INFO 78685 --- [pool-1-thread-1] c.r.p.w.e.a.WorkFlowTaskExecutionAspect  : Before invoking execute() on workflow task name: splunkMonitoringWorkFlowTask
2023-06-27 11:32:10.014  INFO 78685 --- [        task-20] c.r.p.w.e.service.WorkFlowExecutorImpl   : Work report for complexWorkFlow (ID: 4b358d0d-4253-424a-a0d8-5bd0c20b977b): DefaultWorkReport {status=IN_PROGRESS, context=context={WORKFLOW_EXECUTION_SUBWORKFLOWTHREE_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_ID=4b358d0d-4253-424a-a0d8-5bd0c20b977b, WORKFLOW_EXECUTION_SUBWORKFLOWONE_PARENT_WORKFLOW=subWorkFlowTwo, USER_ID=b0f58378-a192-43f9-945d-00eb132f11c3, WORKFLOW_EXECUTION_SUBWORKFLOWTWO_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_EXECUTION_SUBWORKFLOWFOUR_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_ARGUMENTS={domainName=api.com, ipAddress=127.0.0.1}, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_ARGUMENTS={adGroups=adGroupsVALUE, userId=test-user-id}, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_ADGROUPSWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_TASK_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_NAMESPACEWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowTwo, WORKFLOW_EXECUTION_SINGLESIGNONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, WORKFLOW_DEFINITION_NAME=complexWorkFlow, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_ARGUMENTS={projectId=e068417b-37e5-404b-9009-2a2899917a18}, WORKFLOW_EXECUTION_LOADBALANCERWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, PROJECT_ID=3c6cf116-0592-42e4-96e3-15a8a1f7b6ea}}, alertMessage='null, error=''}
2023-06-27 11:32:10.016  INFO 78685 --- [pool-1-thread-1] c.r.p.w.e.a.WorkFlowTaskExecutionAspect  : skipping task: splunkMonitoringWorkFlowTask with status COMPLETED
2023-06-27 11:32:10.016  INFO 78685 --- [pool-1-thread-2] c.r.p.e.c.task.AdGroupsWorkFlowTask      : AdGroupsWorkFlowTask
2023-06-27 11:32:10.016 ERROR 78685 --- [pool-1-thread-2] c.r.p.workflow.task.BaseWorkFlowTask     : parameter userId is not provided for task adGroupsWorkFlowTask!
2023-06-27 11:32:10.017  WARN 78685 --- [pool-1-thread-2] c.r.p.workflow.task.BaseWorkFlowTask     : userId is missing from workContext for workflow task adGroupsWorkFlowTask
2023-06-27 11:32:10.017  WARN 78685 --- [pool-1-thread-2] c.r.p.e.c.task.AdGroupsWorkFlowTask      : Missing keys in context for workflowExecution 3783a426-caef-4ccf-bcc8-583ab02202c8, context: context={WORKFLOW_EXECUTION_SUBWORKFLOWTHREE_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_ID=3783a426-caef-4ccf-bcc8-583ab02202c8, WORKFLOW_EXECUTION_SUBWORKFLOWONE_PARENT_WORKFLOW=subWorkFlowTwo, USER_ID=b0f58378-a192-43f9-945d-00eb132f11c3, WORKFLOW_EXECUTION_SUBWORKFLOWTWO_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONAPPROVALWORKFLOWCHECKERTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_SUBWORKFLOWFOUR_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_ARGUMENTS={domainName=api.com, ipAddress=127.0.0.1}, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_STATUS=FAILED, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_ARGUMENTS={adGroups=adGroupsVALUE}, WORKFLOW_EXECUTION_ADGROUPSWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_TASK_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_NAMESPACEWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowTwo, WORKFLOW_EXECUTION_SINGLESIGNONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, WORKFLOW_DEFINITION_NAME=complexWorkFlow, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_ARGUMENTS={projectId=48def7d9-20aa-4e73-a393-70d350843317}, WORKFLOW_EXECUTION_LOADBALANCERWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, PROJECT_ID=3c6cf116-0592-42e4-96e3-15a8a1f7b6ea}}
2023-06-27 11:32:10.018  INFO 78685 --- [pool-2-thread-1] c.r.p.w.e.a.WorkFlowExecutionAspect      : Workflow subWorkFlowOne is FAILED!
2023-06-27 11:32:10.018  INFO 78685 --- [pool-2-thread-1] c.r.p.workflows.workflow.SequentialFlow  : Work unit "subWorkFlowOne" returned with WorkStatus = "FAILED", skipping subsequent work units
2023-06-27 11:32:10.018  INFO 78685 --- [pool-2-thread-1] c.r.p.w.e.a.WorkFlowExecutionAspect      : Workflow subWorkFlowTwo is FAILED!
2023-06-27 11:32:10.019  INFO 78685 --- [        task-21] c.r.p.w.e.a.WorkFlowExecutionAspect      : Workflow subWorkFlowThree is FAILED!
2023-06-27 11:32:10.020  INFO 78685 --- [        task-21] c.r.p.workflows.workflow.SequentialFlow  : Work unit "subWorkFlowThree" returned with WorkStatus = "FAILED", skipping subsequent work units
2023-06-27 11:32:10.020  INFO 78685 --- [        task-21] c.r.p.w.e.a.WorkFlowExecutionAspect      : Workflow complexWorkFlow is FAILED!
2023-06-27 11:32:10.022  INFO 78685 --- [        task-21] c.r.p.w.e.service.WorkFlowExecutorImpl   : Work report for complexWorkFlow (ID: 3783a426-caef-4ccf-bcc8-583ab02202c8): com.redhat.parodos.workflows.workflow.ParallelFlowReport@691dea9c
2023-06-27 11:32:10.022 ERROR 78685 --- [        task-21] c.r.p.w.e.service.WorkFlowExecutorImpl   : Workflow complexWorkFlow (ID: 3783a426-caef-4ccf-bcc8-583ab02202c8) failed. Check the logs for errors coming from the tasks in this workflow.
2023-06-27 11:32:10.022  INFO 78685 --- [        task-21] c.r.p.w.e.service.WorkFlowExecutorImpl   : execute rollback workflow complexRollbackWorkFlow for workflow complexWorkFlow (ID: 3783a426-caef-4ccf-bcc8-583ab02202c8)
2023-06-27 11:32:10.022  INFO 78685 --- [        task-21] c.r.p.w.engine.WorkFlowEngineImpl        : Running workflow 'complexRollbackWorkFlow'
2023-06-27 11:32:10.022  INFO 78685 --- [        task-21] c.r.p.w.e.a.WorkFlowExecutionAspect      : Before invoking execute() on workflow: complexRollbackWorkFlow with workContext: context={WORKFLOW_EXECUTION_SUBWORKFLOWTHREE_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_ID=3783a426-caef-4ccf-bcc8-583ab02202c8, WORKFLOW_EXECUTION_SUBWORKFLOWONE_PARENT_WORKFLOW=subWorkFlowTwo, USER_ID=b0f58378-a192-43f9-945d-00eb132f11c3, WORKFLOW_EXECUTION_SUBWORKFLOWTWO_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONAPPROVALWORKFLOWCHECKERTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_SUBWORKFLOWFOUR_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_ARGUMENTS={domainName=api.com, ipAddress=127.0.0.1}, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_STATUS=FAILED, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_ARGUMENTS={adGroups=adGroupsVALUE}, WORKFLOW_EXECUTION_ADGROUPSWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_TASK_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_NAMESPACEWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowTwo, WORKFLOW_EXECUTION_SINGLESIGNONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, WORKFLOW_DEFINITION_NAME=complexWorkFlow, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_ARGUMENTS={projectId=48def7d9-20aa-4e73-a393-70d350843317}, WORKFLOW_EXECUTION_LOADBALANCERWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, PROJECT_ID=3c6cf116-0592-42e4-96e3-15a8a1f7b6ea}}
2023-06-27 11:32:10.024  INFO 78685 --- [        task-21] c.r.p.w.e.a.WorkFlowTaskExecutionAspect  : Before invoking execute() on workflow task name: rollbackWorkFlowTask
2023-06-27 11:32:10.025 ERROR 78685 --- [        task-21] c.r.p.w.e.a.WorkFlowExecutionAspect      : Workflow complexRollbackWorkFlow has failed! with error: Cannot invoke "com.redhat.parodos.workflow.execution.entity.WorkFlowExecution.getId()" because "workFlowExecution" is null

2023-06-27 11:32:15.001  INFO 78685 --- [   scheduling-1] c.r.p.w.e.a.WorkFlowExecutionAspect      : Before invoking execute() on workflow: namespaceApprovalWorkFlowChecker with workContext: context={WORKFLOW_EXECUTION_SUBWORKFLOWTHREE_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_ID=4b358d0d-4253-424a-a0d8-5bd0c20b977b, WORKFLOW_EXECUTION_SUBWORKFLOWONE_PARENT_WORKFLOW=subWorkFlowTwo, USER_ID=b0f58378-a192-43f9-945d-00eb132f11c3, WORKFLOW_EXECUTION_SUBWORKFLOWTWO_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_EXECUTION_SUBWORKFLOWFOUR_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_ARGUMENTS={domainName=api.com, ipAddress=127.0.0.1}, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_ARGUMENTS={adGroups=adGroupsVALUE, userId=test-user-id}, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_ADGROUPSWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_TASK_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_NAMESPACEWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowTwo, WORKFLOW_EXECUTION_SINGLESIGNONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, WORKFLOW_DEFINITION_NAME=complexWorkFlow, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_ARGUMENTS={projectId=e068417b-37e5-404b-9009-2a2899917a18}, WORKFLOW_EXECUTION_LOADBALANCERWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, PROJECT_ID=3c6cf116-0592-42e4-96e3-15a8a1f7b6ea}}
2023-06-27 11:32:15.010  INFO 78685 --- [   scheduling-1] c.r.p.w.e.a.WorkFlowTaskExecutionAspect  : Before invoking execute() on workflow task name: namespaceApprovalWorkFlowCheckerTask
2023-06-27 11:32:15.020  INFO 78685 --- [   scheduling-1] c.c.NamespaceApprovalWorkFlowCheckerTask : NamespaceApprovalWorkFlowCheckerTask
2023-06-27 11:32:15.022  INFO 78685 --- [   scheduling-1] c.r.p.w.e.a.WorkFlowExecutionAspect      : Workflow namespaceApprovalWorkFlowChecker is COMPLETED!
2023-06-27 11:32:15.025  INFO 78685 --- [   scheduling-1] r.p.w.e.a.CheckerWorkFlowPostInterceptor : Stop workflow checker: namespaceApprovalWorkFlowChecker schedule
2023-06-27 11:32:15.025  INFO 78685 --- [   scheduling-1] c.r.p.w.e.s.WorkFlowSchedulerServiceImpl : Stopping workflow: namespaceApprovalWorkFlowChecker for project: 3c6cf116-0592-42e4-96e3-15a8a1f7b6ea and user: b0f58378-a192-43f9-945d-00eb132f11c3
2023-06-27 11:32:15.029  INFO 78685 --- [        task-22] c.r.p.w.e.service.WorkFlowExecutorImpl   : Execute workflow complexWorkFlow (ID: 4b358d0d-4253-424a-a0d8-5bd0c20b977b)
2023-06-27 11:32:15.030  INFO 78685 --- [        task-22] c.r.p.w.engine.WorkFlowEngineImpl        : Running workflow 'complexWorkFlow'
2023-06-27 11:32:15.030  INFO 78685 --- [        task-22] c.r.p.w.e.a.WorkFlowExecutionAspect      : Before invoking execute() on workflow: complexWorkFlow with workContext: context={WORKFLOW_EXECUTION_SUBWORKFLOWTHREE_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_TASK_EXECUTION_NAMESPACEAPPROVALWORKFLOWCHECKERTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_ID=4b358d0d-4253-424a-a0d8-5bd0c20b977b, WORKFLOW_EXECUTION_SUBWORKFLOWONE_PARENT_WORKFLOW=subWorkFlowTwo, USER_ID=b0f58378-a192-43f9-945d-00eb132f11c3, WORKFLOW_EXECUTION_SUBWORKFLOWTWO_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_EXECUTION_SUBWORKFLOWFOUR_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_ARGUMENTS={domainName=api.com, ipAddress=127.0.0.1}, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_ARGUMENTS={adGroups=adGroupsVALUE, userId=test-user-id}, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_ADGROUPSWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_TASK_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_NAMESPACEWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowTwo, WORKFLOW_EXECUTION_SINGLESIGNONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, WORKFLOW_DEFINITION_NAME=complexWorkFlow, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_ARGUMENTS={projectId=e068417b-37e5-404b-9009-2a2899917a18}, WORKFLOW_EXECUTION_LOADBALANCERWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, PROJECT_ID=3c6cf116-0592-42e4-96e3-15a8a1f7b6ea}}
2023-06-27 11:32:15.037  INFO 78685 --- [        task-22] c.r.p.w.e.a.WorkFlowExecutionAspect      : Before invoking execute() on workflow: subWorkFlowThree with workContext: context={WORKFLOW_EXECUTION_SUBWORKFLOWTHREE_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_TASK_EXECUTION_NAMESPACEAPPROVALWORKFLOWCHECKERTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_ID=4b358d0d-4253-424a-a0d8-5bd0c20b977b, WORKFLOW_EXECUTION_SUBWORKFLOWONE_PARENT_WORKFLOW=subWorkFlowTwo, USER_ID=b0f58378-a192-43f9-945d-00eb132f11c3, WORKFLOW_EXECUTION_SUBWORKFLOWTWO_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_EXECUTION_SUBWORKFLOWFOUR_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_ARGUMENTS={domainName=api.com, ipAddress=127.0.0.1}, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_ARGUMENTS={adGroups=adGroupsVALUE, userId=test-user-id}, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_ADGROUPSWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_TASK_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_NAMESPACEWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowTwo, WORKFLOW_EXECUTION_SINGLESIGNONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, WORKFLOW_DEFINITION_NAME=complexWorkFlow, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_ARGUMENTS={projectId=e068417b-37e5-404b-9009-2a2899917a18}, WORKFLOW_EXECUTION_LOADBALANCERWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, PROJECT_ID=3c6cf116-0592-42e4-96e3-15a8a1f7b6ea}}
2023-06-27 11:32:15.043  INFO 78685 --- [pool-2-thread-2] c.r.p.w.e.a.WorkFlowTaskExecutionAspect  : Before invoking execute() on workflow task name: sslCertificationWorkFlowTask
2023-06-27 11:32:15.043  INFO 78685 --- [pool-2-thread-1] c.r.p.w.e.a.WorkFlowExecutionAspect      : Before invoking execute() on workflow: subWorkFlowTwo with workContext: context={WORKFLOW_EXECUTION_SUBWORKFLOWTHREE_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_TASK_EXECUTION_NAMESPACEAPPROVALWORKFLOWCHECKERTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_ID=4b358d0d-4253-424a-a0d8-5bd0c20b977b, WORKFLOW_EXECUTION_SUBWORKFLOWONE_PARENT_WORKFLOW=subWorkFlowTwo, USER_ID=b0f58378-a192-43f9-945d-00eb132f11c3, WORKFLOW_EXECUTION_SUBWORKFLOWTWO_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_EXECUTION_SUBWORKFLOWFOUR_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_ARGUMENTS={domainName=api.com, ipAddress=127.0.0.1}, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_ARGUMENTS={adGroups=adGroupsVALUE, userId=test-user-id}, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_ADGROUPSWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_TASK_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_NAMESPACEWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowTwo, WORKFLOW_EXECUTION_SINGLESIGNONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, WORKFLOW_DEFINITION_NAME=complexWorkFlow, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_ARGUMENTS={projectId=e068417b-37e5-404b-9009-2a2899917a18}, WORKFLOW_EXECUTION_LOADBALANCERWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, PROJECT_ID=3c6cf116-0592-42e4-96e3-15a8a1f7b6ea}}
2023-06-27 11:32:15.050  INFO 78685 --- [pool-2-thread-1] c.r.p.w.e.a.WorkFlowExecutionAspect      : Before invoking execute() on workflow: subWorkFlowOne with workContext: context={WORKFLOW_EXECUTION_SUBWORKFLOWTHREE_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_TASK_EXECUTION_NAMESPACEAPPROVALWORKFLOWCHECKERTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_ID=4b358d0d-4253-424a-a0d8-5bd0c20b977b, WORKFLOW_EXECUTION_SUBWORKFLOWONE_PARENT_WORKFLOW=subWorkFlowTwo, USER_ID=b0f58378-a192-43f9-945d-00eb132f11c3, WORKFLOW_EXECUTION_SUBWORKFLOWTWO_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_EXECUTION_SUBWORKFLOWFOUR_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_ARGUMENTS={domainName=api.com, ipAddress=127.0.0.1}, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_ARGUMENTS={adGroups=adGroupsVALUE, userId=test-user-id}, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_ADGROUPSWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_TASK_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_NAMESPACEWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowTwo, WORKFLOW_EXECUTION_SINGLESIGNONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, WORKFLOW_DEFINITION_NAME=complexWorkFlow, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_ARGUMENTS={projectId=e068417b-37e5-404b-9009-2a2899917a18}, WORKFLOW_EXECUTION_LOADBALANCERWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, PROJECT_ID=3c6cf116-0592-42e4-96e3-15a8a1f7b6ea}}
2023-06-27 11:32:15.050  INFO 78685 --- [pool-2-thread-2] c.r.p.w.e.a.WorkFlowTaskExecutionAspect  : skipping task: sslCertificationWorkFlowTask with status COMPLETED
2023-06-27 11:32:15.053  INFO 78685 --- [pool-2-thread-1] c.r.p.w.e.a.WorkFlowExecutionAspect      : skipping workflow: subWorkFlowOne is already completed
2023-06-27 11:32:15.053  INFO 78685 --- [pool-2-thread-1] c.r.p.w.e.a.WorkFlowTaskExecutionAspect  : Before invoking execute() on workflow task name: namespaceWorkFlowTask
2023-06-27 11:32:15.058  INFO 78685 --- [pool-2-thread-1] c.r.p.w.e.a.WorkFlowTaskExecutionAspect  : skipping task: namespaceWorkFlowTask with status COMPLETED
2023-06-27 11:32:15.058  INFO 78685 --- [pool-2-thread-1] c.r.p.w.e.a.WorkFlowExecutionAspect      : Workflow subWorkFlowTwo is COMPLETED!
2023-06-27 11:32:15.060  INFO 78685 --- [        task-22] c.r.p.w.e.a.WorkFlowExecutionAspect      : Workflow subWorkFlowThree is COMPLETED!
2023-06-27 11:32:15.060  INFO 78685 --- [        task-22] entInfrastructureWorkFlowPostInterceptor : workflow: subWorkFlowThree has a pending/running checker: checker is pending in execution: 4b358d0d-4253-424a-a0d8-5bd0c20b977b
2023-06-27 11:32:15.062  INFO 78685 --- [        task-22] c.r.p.workflows.workflow.SequentialFlow  : Work unit "subWorkFlowThree" returned with WorkStatus = "IN_PROGRESS", skipping subsequent work units
2023-06-27 11:32:15.062  INFO 78685 --- [        task-22] c.r.p.w.e.a.WorkFlowExecutionAspect      : Workflow complexWorkFlow is IN_PROGRESS!
2023-06-27 11:32:15.066  INFO 78685 --- [        task-22] c.r.p.w.e.service.WorkFlowExecutorImpl   : Work report for complexWorkFlow (ID: 4b358d0d-4253-424a-a0d8-5bd0c20b977b): DefaultWorkReport {status=IN_PROGRESS, context=context={WORKFLOW_EXECUTION_SUBWORKFLOWTHREE_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_TASK_EXECUTION_NAMESPACEAPPROVALWORKFLOWCHECKERTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_ID=4b358d0d-4253-424a-a0d8-5bd0c20b977b, WORKFLOW_EXECUTION_SUBWORKFLOWONE_PARENT_WORKFLOW=subWorkFlowTwo, USER_ID=b0f58378-a192-43f9-945d-00eb132f11c3, WORKFLOW_EXECUTION_SUBWORKFLOWTWO_PARENT_WORKFLOW=subWorkFlowThree, WORKFLOW_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_EXECUTION_SUBWORKFLOWFOUR_PARENT_WORKFLOW=complexWorkFlow, WORKFLOW_TASK_EXECUTION_SSLCERTIFICATIONWORKFLOWTASK_ARGUMENTS={domainName=api.com, ipAddress=127.0.0.1}, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_TASK_EXECUTION_ADGROUPSWORKFLOWTASK_ARGUMENTS={adGroups=adGroupsVALUE, userId=test-user-id}, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_ADGROUPSWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowOne, WORKFLOW_TASK_EXECUTION_SPLUNKMONITORINGWORKFLOWTASK_STATUS=COMPLETED, WORKFLOW_EXECUTION_NAMESPACEWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowTwo, WORKFLOW_EXECUTION_SINGLESIGNONWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, WORKFLOW_DEFINITION_NAME=complexWorkFlow, WORKFLOW_TASK_EXECUTION_NAMESPACEWORKFLOWTASK_ARGUMENTS={projectId=e068417b-37e5-404b-9009-2a2899917a18}, WORKFLOW_EXECUTION_LOADBALANCERWORKFLOWTASK_PARENT_WORKFLOW=subWorkFlowFour, PROJECT_ID=3c6cf116-0592-42e4-96e3-15a8a1f7b6ea}}, alertMessage='null, error=''}

```

I will create a ticket to investigate further

Reply:
https://issues.redhat.com/browse/FLPATH-499

Reply:
> strange indeed, especially since the expected status of the workflow has been reached.

Yep, the workflow fails as expected, then next test is executed but the rollback workflow of the previous test is still being executed and fails while the current test has its workflow stuck in IN_PROGRESS